src/main/java/core/basesyntax/strategy/operation/PurchaseOperationHandler.java

```diff
@@ -0,0 +1,8 @@
+package core.basesyntax.strategy.operation;
+
+public class PurchaseOperationHandler implements OperationHandler {
+    @Override
+    public int getQuantity(int prev, int value) {
+        return prev - value;
+    }
```

Comment:
Handling Purchase operation.
Check result balance before saving it in the Storage - it should be positive. Throw `RuntimeException` in case the balance is negative. Currently, the `getQuantity` method doesn't check if the result of the purchase operation (prev - value) is negative, which could lead to incorrect behavior if a purchase is made that exceeds the available quantity.